src/lakefs_spec/errors.py

```diff
@@ -8,61 +8,67 @@
 from __future__ import annotations
 
 import errno
-import functools
 import json
-from typing import Any
+from functools import partial
+from urllib.error import HTTPError
 
 from lakefs_sdk import ApiException
 
-HTTP_CODE_TO_ERROR: dict[int, type[OSError]] = {
+HTTP_CODE_TO_ERROR: dict[int, type[OSError] | partial[OSError]] = {
+    400: partial(IOError, errno.EINVAL),
     401: PermissionError,
     403: PermissionError,
     404: FileNotFoundError,
+    410: FileNotFoundError,  # Gone (temporarily / permanently unavailable)
+    416: partial(IOError, errno.EINVAL),  # invalid range
+    420: partial(IOError, errno.EBUSY),  # too many requests
```

Comment:
We have to be a bit careful with the `errno` package since it might be platform-dependent. From the docs:

> Of the following list, symbols that are not used on the current platform are not defined by the module.

We should verify that these error numbers are in fact defined on all supported platforms (Linux/Darwin are easy enough, Windows I'm not so sure about)

Reply:
I went with the corresponding `s3fs` implementation, it seems none of the codes we use are affected: https://github.com/fsspec/s3fs/blob/main/s3fs/errors.py